utils/dyn-config.c

```diff
@@ -293,6 +294,22 @@ glusterBlockReadConfig(gbConfig *cfg, ssize_t *len)
     *len += m;
   }
 
+  /*
+   * In-case if the editor (vim) follows write to a new file (.swp, .tmp ..)
+   * and move it to actual file name later. There is a window, where we will
+   * encounter one case that the file data is not flushed to the disk, so in
+   * another process(here) when reading it will be empty.
+   *
+   * Let just wait and try again.
+   */
+  if (*len == 0) {
+    LOG("mgmt", GB_LOG_DEBUG,
+        "The pagecache is not flushed yet, will try to read it again");
+    fclose(fp);
+    sleep(1);
+    goto retry;
```

Comment:
@lxbsz, should there be some limit to this retry ?
May be 5 times like we do in https://github.com/open-iscsi/tcmu-runner/pull/506/commits/f7b766b5e595ade619a023a4b1e1f0e1754c6d39 ?

Reply:
Yeah, make sense.